packages/server/src/server.ts

```diff
@@ -217,6 +218,18 @@ app.useRpc('eth_blockNumber', async () => {
  * returns: Gas used - hex encoded integer
  */
 app.useRpc('eth_estimateGas', async (params: any) => {
+  const validationError = Validator.validateParams(params, {
```

Comment:
Validating too early may prevent the log and metrics of the endpoint call from occurring.
Please check this is not the case.
if it is maybe you want to pass the Validator into the `logAndHandleResponse()` so that it gets run before the service logic is called

Reply:
If the validation happens in `logAndHandleResponse()`, we will only get part of the logs 
```
logger.debug(`${requestIdPrefix} ${methodName}`);
```
because trace logging the called method and the params passed
```
this.logger.trace(`${requestIdPrefix} getBlockByHash(hash=${hash}, showDetails=${showDetails})`);
```
happens in function of each method the `eth.ts` module in the `relay` package.
Would this be enough?

Reply:
Good question.  Let me ask a question in response.  Does the validation error message give you enough information about all of the parameters were passed into the function and which one was failing validation?  If so, perhaps there is enough information for the reader of the log to piece together what went wrong?

One alternate suggestion which is kind of ugly as it involves a fair amount of refactoring is to pass in an object instead of the `requestId` string.  The object could contain the `validator` and the `requestId`.  I'm not sure what a good name for such an object might be.

Reply:
It either gives you the index of the parameter, for example in
```
{ "method" : "eth_getBlockByNumber" ,"params" : ["1", false], "id": 1, "jsonrpc": "2.0" }
```
It will return that `param at index 0` is not a hex number.

In the case where one of the params itself is an object (Filter or Transaction)
```
{ "method" : "eth_getLogs", "params" : [ {"address": "0x1234"} ], "id":1 , "jsonrpc":"2.0" }
```
It will return that the `address` property of the `Filter` object is not a valid address